CMakeLists.txt

```diff
@@ -221,6 +221,13 @@ else()
   add_definitions(-Wall -Wextra -pedantic -Wno-unused-parameter
     -Wstrict-prototypes -std=gnu99)
 
+  if(CMAKE_COMPILER_IS_GNUCC)
```

Comment:
We can't check for something if we don't know what it will look like.  For example, GCC's "ignore UBSAN" looks like `__attribute__((no_sanitize_undefined))` but clang's looks like `__attribute__((no_sanitize("undefined")))`.

I suppose this check should be for the presence of the attribute, rather than the compile flag, though.